SpatialGDK/ExternalSchemaCodegen/Programs/CodeGenerator/Program.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using CommandLine;
+using Improbable.Codegen.Base;
+using Improbable.CodeGen.Base;
+using Improbable.CodeGen.Unreal;
+
+namespace CodeGenerator
+{
+    public class Options
+    {
+        [Option("input-bundle", Required = true,
+            HelpText = "The path to the JSON Bundle file output by the SpatialOS schema_compiler.")]
+        public string InputBundle { get; set; }
+
+        [Option("output-dir", Required = true,
+            HelpText = "The path to write the generated code to.")]
+        public string OutputDir { get; set; }
+    }
+
+    internal class Program
+    {
+        private static void Main(string[] args)
+        {
+            Parser.Default.ParseArguments<Options>(args)
+                .WithParsed(Run)
+                .WithNotParsed(errors =>
+                {
+                    foreach (var error in errors)
+                    {
+                        Console.Error.WriteLine(error);
+                    }
+
+                    Environment.ExitCode = 1;
+                });
+        }
+
+        private static void Run(Options options)
+        {
+            try
+            {
+                var bundle = SchemaBundleLoader.LoadBundle(options.InputBundle);
+
+                var generators = new List<ICodeGenerator>
```

Comment:
What's the idea here with the list of code generators? Are there any plans to expose further functionality to the user?

Reply:
So this is based off the ecosystems code generation code that Jared has been working on. The idea is that you can mixin various different generators depending on your use case. For the GDK it may be simpler just to collapse this notion, or maybe preferable to keep the model more similar to others, I'm somewhat indifferent.